core/src/test/scala/com/stripe/dagon/DataFlowTest.scala

```diff
@@ -23,6 +24,66 @@ object DataFlowTest {
 
     def tagged[A](a: A): Flow[T] =
       Flow.Tagged(this, a)
+
+    /*
+     * For large dags you need to eagerly cache hashcode
+     * or you will stack overflow
+     */
+    override val hashCode = scala.util.hashing.MurmurHash3.productHash(this)
+
+    /*
+     * You need a custom equals to avoid stack overflow
+     */
+    override def equals(that: Any) = {
```

Comment:
We probably want to give our user some easier way to define this or avoid using `equals` and `hashCode` in our code.